src/util/simplify_expr.cpp

```diff
@@ -1818,11 +1815,10 @@ bool simplify_exprt::simplify_byte_extract(byte_extract_exprt &expr)
         op_type_ptr->id()==ID_array;
         op_type_ptr=&(ns.follow(*op_type_ptr).subtype()))
     {
-      // no arrays of zero-sized objects
-      assert(el_size>0);
-      // no arrays of non-byte sized objects
-      assert(el_size%8==0);
-      mp_integer el_bytes=el_size/8;
+      DATA_INVARIANT(*el_size>0, "no arrays of zero-sized objects");
```

Comment:
Rephrase as a "should" statement for clarity when printed as an error dump (as written it's not clear whether we're forbidding or complaining that no such array was found)